lib/Doctrine/Enumeration/Exception/NotSupportedException.php

```diff
@@ -0,0 +1,9 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Enumeration\Exception;
+
+final class NotSupportedException extends \LogicException implements \Throwable
```

Comment:
IMO there should be at least another exception if not 2 for invalid values / names, because those will be caught a lot, and should be distinguishable from using a magic method that is not supported.